plugin/pulpcore/plugin/stages/declarative_version.py

```diff
@@ -86,10 +87,18 @@ def __init__(self, first_stage, repository, sync_mode='mirror'):
         Raises:
             ValueError: if 'sync_mode' is passed an invalid value.
         """
-
         if sync_mode != 'mirror' and sync_mode != 'additive':
             msg = _("'sync_mode' must either be 'mirror' or 'additive' not '{sync_mode}'")
             raise ValueError(msg.format(sync_mode=sync_mode))
+
+        self.inject_before = inject_before or []
+        self.inject_after = inject_after or []
+
+        if not isinstance(self.inject_before, list):
```

Comment:
This check will fail for other iterable types, e.g. tuples. Can we instead check if it's a subclass of Stage and if it isn't then listify it?

Reply:
I know, but what if there's just a single tuple, fox example `inject_before=('ContentUnitSaver', 'BeforeContentUnitSaver')`. Then the for loops in the `create()` will failed.

Reply:
Yup I had forgotten the data format. :) Let's keep this exactly how you have it. Please disregard my request. Thanks for the info.

Reply:
Allright then. :)